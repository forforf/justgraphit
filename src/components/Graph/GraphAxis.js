import React, {useEffect, useRef} from 'react';


function GraphAxis({axis, bottom, left, d3select}) {
  const axisRef = useRef(null);

  // triggered on render or changes to axis props.
  useEffect(() => {
    d3select(axisRef.current).call(axis)
  }, [axis, d3select]);

  const translate = `translate(${left},${bottom})`;

  return (
    <g className="axis" ref={axisRef} transform={translate} >
    </g>
  );
}

//ToDo: propTypes
// GraphAxis.propTypes = {
//   height:React.PropTypes.number,
//   axis:React.PropTypes.func,
//   axisType:React.PropTypes.oneOf(['x','y'])
// };

export default GraphAxis;
