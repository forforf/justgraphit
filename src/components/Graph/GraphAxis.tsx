import React, { useEffect, useRef } from 'react';
import { D3Select } from '../../JustGraphitTypes';

type AxisType = d3.Axis<Date> | d3.Axis<number>;

type Props<T extends AxisType> = {
  axis: T;
  bottom: number;
  left: number;
  d3select: D3Select;
};

function GraphAxis<T extends AxisType>({
  axis,
  bottom,
  left,
  d3select,
}: Props<T>): JSX.Element {
  const axisRef = useRef(null);

  // triggered on render or changes to axis props.
  useEffect(() => {
    if (axisRef !== null && axisRef.current) {
      d3select(axisRef.current).call(axis);
    }
  }, [axis, d3select]);

  const translate = `translate(${left},${bottom})`;

  return <g className="axis" ref={axisRef} transform={translate} />;
}

export default GraphAxis;
