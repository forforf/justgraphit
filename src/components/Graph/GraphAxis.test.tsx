import React from 'react';
import renderer from 'react-test-renderer';
import GraphAxis from './GraphAxis';

let mockD3select: () => jest.Mock<unknown, unknown[]>;

beforeEach(() => {
  mockD3select = () => jest.fn(() => 'd3.select(...).call(axis) result');
});

test('deep snapshot x-axis', () => {
  const props = {
    axis: jest.fn() as unknown as d3.Axis<number>,
    bottom: 0,
    left: 0,
    d3select: mockD3select,
  };

  const component = renderer.create(<GraphAxis<d3.Axis<number>> {...props} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('deep snapshot y-axis', () => {
  const props = {
    axis: jest.fn() as unknown as d3.Axis<number>,
    bottom: 0,
    left: 0,
    d3select: mockD3select,
  };
  const component = renderer.create(<GraphAxis {...props} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
