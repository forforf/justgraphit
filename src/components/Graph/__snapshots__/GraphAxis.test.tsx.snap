// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deep snapshot x-axis 1`] = `
<g
  className="axis"
  transform="translate(0,0)"
/>
`;

exports[`deep snapshot y-axis 1`] = `
<g
  className="axis"
  transform="translate(0,0)"
/>
`;
